### PR TITLE
Add rosnode and rostopic examples

### PR DESCRIPTION
**Pull request by vmayoral**

### CODE REVIEW

File: rclpy_examples/rostopic.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import rclpy
+from rclpy.qos import qos_profile_default
+from std_msgs.msg import String
+
+import rclpy.rostopic as rostopic
+
+
+def main(args=None):
+    # TODO discuss whether rclpy should or shouldn't be initialized here
+    #rclpy.init(argv)
```
Comment (wjwwood) [inline rclpy_examples/rostopic.py]: It must be.

File: rcl_examples/src/get_node_names.cpp [hunk 2]
```diff
@@ -0,0 +1,38 @@
+#include <rcl/rcl.h>
```
Comment (wjwwood) [inline rcl_examples/src/get_node_names.cpp]: No copyright in this file.

File: rcl_examples/src/get_node_names.cpp [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    
+    rcl_strings_t node_names;
+
+    node_names = rcl_get_zero_initialized_strings();
+
+    rcl_ret_t ret3 = rcl_get_node_names(&node_names);
+    //std::cout << "ret: " << ret3 << std::endl;
+
+    for(unsigned int i = 0; i < node_names.count; i++){
+        std::cout << "->/" << node_names.data[i] << std::endl;
```
Comment (wjwwood) [inline rcl_examples/src/get_node_names.cpp]: We try to use `printf` everywhere to avoid interleaved output (`cout` can flush between `<<`'s).

**Comment by mikaelarguedas** [comment]
closing this in favor or the new commandline tool hosted at https://github.com/ros2/ros2cli/

**Pull request closed without merging**